### DISCUSSION TITLE
Cannot extract some entities when traverse in msp

### DISCUSSION THREAD

**Discussion by En0427** [mozman/ezdxf#1087] [Q&A]
When I was traversing the entities in msp, some entities were not extracted. In my dxf file, these entities in the following image belong to the layer of "00-备注信息不打印".
![Expected](https://github.com/mozman/ezdxf/assets/125962979/fb9514fd-d1a3-4b94-9432-51c9716c0699)
But when I ran the following code, there was no result. May I ask how to solve it. 
`doc = ezdxf.readfile('test.dxf')
msp = doc.modelspace()
for entity in msp:
    if entity.dxf.layer == '00-备注信息不打印图层':
        print(entity.dxftype())`

I checked the layer table and the status of the layer, and the return seems to be fine
`layers = doc.tables.layers.entries.keys()
print(layers)

my = doc.layers.get('00-备注信息不打印图层')
print(my.is_on())
print(my.dxf.name)
print(my.dxf.plot)`
![return](https://github.com/mozman/ezdxf/assets/125962979/b5a69556-39d8-42a0-9e48-7de3e80d4100)

This is the relevant file and code: 
[ralated file.zip](https://github.com/mozman/ezdxf/files/15401623/ralated.file.zip)

**Comment by mozman** [comment]
This entity is inside a block:

![image](https://github.com/mozman/ezdxf/assets/4230966/a52d2e44-826c-450a-87ba-e6a52f9ace73)
